core/os/os_linux.odin

```diff
@@ -253,6 +256,29 @@ X_OK :: 1; // Test for execute permission
 W_OK :: 2; // Test for write permission
 R_OK :: 4; // Test for read permission
 
+passwd :: struct {
+	pw_name: cstring,
+	pw_passwd: cstring,
+	pw_uid: uid_t,
+	pw_gid: gid_t,
+	pw_gecos: cstring,
+	pw_dir: cstring,
+	pw_shell: cstring,
+}
+
+DIR :: opaque [0]byte;
+ino_t :: distinct c.ulong;
+off_t :: distinct c.long;
+
+dirent :: struct {
+	d_ino: ino_t, // Inode number of the file
+	d_off: off_t,
+	d_reclen: c.ushort, // Size in bytes of the returned record
+	d_type: c.uchar, // File type
+
+	d_name: [256]c.char, // Name of file, null-terminated, can exceede 256 chars (in which case d_reclen exceedes the size of this struct)
+}
+
 SYS_GETTID: Syscall : 186;
 
 foreign libc {
```

Comment:
Odin wants to be able to use `-no-crt` on Linux, and to do that usefully we need to remove all usage of libc from the stdlib so that the stdlib may still be used.
Maybe we might want to still provide like a `core:libc` package or something, but. 🤔 
Or maybe we just want to do the syscall _ourselves_ rather than relying on libc - similar to how we do `gettid`, except using inline asm, when we have it.

For now, it's probably useful to still add the new libc things here, but we want to remove them eventually - maybe when we get inline asm, or maybe put them in their own package, like I say.

Anyway, maybe we should add a comment to this foreign block along the lines of:
```
// TODO: Remove dependance on libc. Waiting on inline asm for syscalls?
```

Reply:
Ok, thanks.

Reply:
However, I'm confused... the C package isn't libc. It's just a file defining the C types to their corresponding Odin types, isn't it?

Reply:
Btw, libc is definitely still useful atm until it can **fully** be replaced, because I need it for my new project (I'm rewriting my shell, Paled, in Odin). These functions happened to be additional syscalls that I needed for that.